networking.tf

```diff
@@ -0,0 +1,98 @@
+resource "aws_vpc" "main" {
+    cidr_block = "172.31.0.0/16"
+}
+
+resource "aws_subnet" "public_a" {
+  vpc_id                  = aws_vpc.main.id
+  availability_zone       = "eu-west-1a"
+  cidr_block              = "172.31.0.0/20"
+  map_public_ip_on_launch = true
+}
+
+resource "aws_subnet" "public_b" {
+  vpc_id                  = aws_vpc.main.id
+  availability_zone       = "eu-west-1b"
+  cidr_block              = "172.31.16.0/20"
+  map_public_ip_on_launch = true
+}
+
+resource "aws_subnet" "public_c" {
+  vpc_id                  = aws_vpc.main.id
+  availability_zone       = "eu-west-1c"
+  cidr_block              = "172.31.32.0/20"
+  map_public_ip_on_launch = true
+}
+
+resource "aws_internet_gateway" "main" {
+  vpc_id = aws_vpc.main.id
+}
+
+resource "aws_default_security_group" "main" {
+  vpc_id = aws_vpc.main.id
+
+  ingress {
+    from_port         = 0
+    to_port           = 0
+    protocol          = "-1"
+    self              = true
+  }
+
+  egress {
+    from_port         = 0
+    to_port           = 0
+    protocol          = "-1"
+    cidr_blocks       = [
+        "0.0.0.0/0",
+    ]
+  }
+}
+
+resource "aws_security_group_rule" "main_ingress" {
+  security_group_id = aws_default_security_group.main.id
+  type              = "ingress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  self              = true
+}
+
+resource "aws_security_group_rule" "main_egress" {
+  security_group_id = aws_default_security_group.main.id
+  type              = "egress"
+  from_port         = 0
+  to_port           = 0
+  protocol          = "-1"
+  cidr_blocks       = [
+      "0.0.0.0/0",
+  ]
+}
+
+resource "aws_default_network_acl" "main" {
+  default_network_acl_id = aws_vpc.main.default_network_acl_id
+  subnet_ids = [
+    aws_subnet.public_a.id,
+    aws_subnet.public_b.id,
+    aws_subnet.public_c.id,
+  ]
+
+  ingress {
```

Comment:
By default Terraform removes allow in ingress and egress from default network acl. I've put them here explicitly, later on we could refine on security here.